packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/PersistedNameTests.kt

```diff
@@ -0,0 +1,370 @@
+            value = bsonObjectId
+        )
+    }
+
+    @Test
+    fun query_byPrimaryKeyPersistedName() {
+        realm.writeBlocking {
+            copyToRealm(PersistedNameSample())
+        }
+
+        assertCanQuerySingle(
+            property = PersistedNameSample::publicNamePrimaryKey,
+            nameToQueryBy = "persistedNamePrimaryKey",
+            value = bsonObjectId
+        )
+    }
+
+    @Test
+    fun query_byPrimaryKeyPublicName() {
+        realm.writeBlocking {
+            copyToRealm(PersistedNameSample())
+        }
+
+        assertCanQuerySingle(
+            property = PersistedNameSample::publicNamePrimaryKey,
+            nameToQueryBy = "publicNamePrimaryKey",
+            value = bsonObjectId
+        )
+    }
+
+    @Test
+    fun query_byEmojiPersistedName() {
+        realm.writeBlocking {
+            copyToRealm(PersistedNameSample())
+        }
+
+        assertCanQuerySingle(
+            property = PersistedNameSample::publicNameWithoutEmoji,
+            nameToQueryBy = "persistedNameWithEmoji😊",
+            value = "Realm"
+        )
+    }
+
+    @Test
+    fun dynamicRealmQuery_getValueByPersistedName() {
+        realm.writeBlocking {
+            copyToRealm(PersistedNameSample())
+        }
+
+        val dynamicSample = realm.asDynamicRealm().query(PersistedNameSample::class.simpleName!!)
+            .find()
+            .single()
+
+        assertNotNull(dynamicSample)
+        assertEquals("Realm", dynamicSample.getValue("persistedNameStringField"))
+        assertFailsWithMessage<IllegalArgumentException>("Schema for type '${PersistedNameSample::class.simpleName!!}' doesn't contain a property named 'publicNameStringField'") {
+            dynamicSample.getValue("publicNameStringField")
+        }
+    }
+
+    // --------------------------------------------------
+    // Backlinks
+    // --------------------------------------------------
+
+    @Test
+    fun backlinks_canPointToPersistedName() {
+        val config = RealmConfiguration
+            .Builder(schema = setOf(PersistedNameParentSample::class, PersistedNameChildSample::class))
+            .name("backlinks.realm")
+            .directory("$tmpDir/foo")
+            .build()
+        val realm = Realm.open(config)
+
+        realm.writeBlocking {
+            // Add a child with 5 parents
+            val child = copyToRealm(PersistedNameChildSample())
+            val parents = Array(5) {
+                this.copyToRealm(PersistedNameParentSample(it))
+            }
+            assertEquals(0, child.parents.size)
+            parents.forEach { parent ->
+                parent.publicNameChildField = child
+            }
+        }
+
+        val queriedChild = realm.query<PersistedNameChildSample>()
+            .find()
+            .single()
+
+        assertEquals(5, queriedChild.parents.size)
+        assertEquals(1, queriedChild.parents.query("id = 3").find().size)
+    }
```

Comment:
I miss a test to validate that persisted names are accessible via named backlink queries: https://www.mongodb.com/docs/realm/realm-query-language/#backlink-queries